aten/src/ATen/functorch/BatchRulesBinaryOps.cpp

```diff
@@ -287,6 +287,24 @@ void fill__Tensor_batch_rule(
   std::get<0>(self_and_other).copy_(std::get<1>(self_and_other));
 }
 
+std::tuple<Tensor,optional<int64_t>> fill_Tensor_batch_rule(
+    Tensor& self,
+    optional<int64_t> self_bdim,
+    const Tensor& other,
+    optional<int64_t> other_bdim) {
+  if (!other_bdim.has_value()) {
+    // Optimization: fill is faster than the other path which does
+    // reshaping + copy_
+    return std::make_tuple(at::fill(self, other), self_bdim);
+  }
+  auto self_and_other = _binary_pointwise_helper(
+      self, self_bdim, other, other_bdim, /*do_type_promotion*/false);
+  auto self_and_other_broadcasted = at::broadcast_tensors(
+      {std::get<0>(self_and_other), std::get<1>(self_and_other)});
+  auto result = at::empty_like(self_and_other_broadcasted[0]);
+  return std::make_tuple(result.copy_(self_and_other_broadcasted[1]), 0);
```

Comment:
We cannot use a BINARY_POINTWISE macro for at::fill because at::fill(self, other) requires other to be a scalar tensor.

Reply:
Just thinking longer term in PT2/compile world: the extra empty_like() + copy_() feels a bit bad, because the copy_() will be functionalized and we'll get `empty_like()` + `copy()` in the graph, but I think we can reliably expect e.g. inductor to optimize it away.

Reply:
If we wanted to avoid it, we probably could use repeat? The logic might get a little messier though so not sure if it's worth it

Reply:
Yeah, repeat or expand+contiguous on self_and_other_broadcasted[1] might be better. I'll go think about it for a bit, the logic might not be too bad because we restrict `other` to having at most 2 dimensions